### PR TITLE
[chiptest] Add support for door lock app tests

### PR DESCRIPTION
**Pull request by arkq**
#### Problem

Door lock tests are skipped when running tests with `./scripts/tests/run_test_suite.py`.

#### Change overview

* Added door lock application to `run` command (`--door-lock-app`)
* Fixed PEP8 warnings in chiptest Python files 

#### Testing

Running test suite on Linux:
```sh
./scripts/tests/run_test_suite.py \
    --target-glob "DL_*" \
    --chip-tool out/examples/chip-tool/chip-tool \
    run \
      --all-clusters-app out/examples/all-clusters-app/chip-all-clusters-app  \
      --door-lock-app out/examples/door-lock-app/chip-door-lock-app \
      --tv-app out/examples/tv-app/chip-tv-app
```

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -195,7 +197,7 @@ jobs:
                   ./scripts/run_in_build_env.sh \
                   "./scripts/tests/run_test_suite.py \
                      --chip-tool ./out/darwin-x64-chip-tool-${BUILD_VARIANT}${CHIP_TOOL_VARIANT}/chip-tool \
-                     --target-skip-glob '{TestGroupMessaging,TV_*}' \
+                     --target-skip-glob '{TestGroupMessaging,DL_*,TV_*}' \
```
Comment (bzbarsky-apple) [inline .github/workflows/tests.yaml:200]: Would be better to build door lock app and run the tests against it here, no?

Comment (Arkq) [reply .github/workflows/tests.yaml:200]: I thought that since TV tests were already excluded, DL tests should also be excluded. But I can change that.

Comment (bzbarsky-apple) [reply .github/workflows/tests.yaml:200]: I have no idea why the TV tests are excluded either, fwiw...

**Comment by woody-apple** [comment]
Fast tracking, given this has had enough time for review.

**Comment by Arkq** [comment]
@woody-apple I think that this PR will have to wait a little bit more. After adding door lock app to the test suite, an error has emerged. There is a use after free error reported by clang thread analyses. Currently, I'm trying to resolve this use after free issue. Maybe I should convert this PR into a PR draft?

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (Arkq) [reply .github/workflows/tests.yaml]: I've enabled these test for Darwin platform, unfortunately there is a fail like that:
```
2022-03-04 11:[48](https://github.com/project-chip/connectedhomeip/runs/5421437150?check_suite_focus=true#step:9:48):29.[50](https://github.com/project-chip/connectedhomeip/runs/5421437150?check_suite_focus=true#step:9:50)7 INFO    Each test will be executed 1 times
2022-03-04 11:48:29.508 INFO    Starting iteration 1
2022-03-04 11:48:39.607 ERROR   !!!!!!!!!!!!!!!!!!!! ERROR !!!!!!!!!!!!!!!!!!!!!!
2022-03-04 11:48:39.608 ERROR   ================ CAPTURED LOG START ==================
2022-03-04 11:48:39.608 ERROR   11:48:29.509 - APP       : EXECUTING ['./out/darwin-x64-door-lock-no-ble-asan/chip-door-lock-app', '--discriminator', '[57](https://github.com/project-chip/connectedhomeip/runs/5421437150?check_suite_focus=true#step:9:57)7', '--interface-id', '-1']
2022-03-04 11:48:39.[60](https://github.com/project-chip/connectedhomeip/runs/5421437150?check_suite_focus=true#step:9:60)9 ERROR   ================ CAPTURED LOG END ====================
2022-03-04 11:48:39.609 ERROR   DL_LockUnlock - FAILED in 10.10 seconds
Traceback (most recent call last):
  File "./scripts/tests/run_test_suite.py", line 213, in cmd_run
    test.Run(runner, apps_register, paths)
  File "/Users/runner/work/connectedhomeip/connectedhomeip/scripts/tests/chiptest/test_definition.py", line 232, in Run
    app.start(str(randrange(1, 4096)))
  File "/Users/runner/work/connectedhomeip/connectedhomeip/scripts/tests/chiptest/test_definition.py", line 43, in start
    self.waitForAnyAdvertisement(process, outpipe)
  File "/Users/runner/work/connectedhomeip/connectedhomeip/scripts/tests/chiptest/test_definition.py", line 76, in waitForAnyAdvertisement
    self.__waitFor("mDNS service published:", process, outpipe)
  File "/Users/runner/work/connectedhomeip/connectedhomeip/scripts/tests/chiptest/test_definition.py", line 125, in __waitFor
    raise Exception('Timeout while waiting for %s' % waitForString)
Exception: Timeout while waiting for mDNS service published:
127.0.0.1 - - [04/Mar/2022 11:48:39] "POST / HTTP/1.1" 200 -
Error: The action has timed out.
```

I'm not able to test it on macOS on my side, so I will simply keep this skip glop for now.

Comment (bzbarsky-apple) [reply .github/workflows/tests.yaml:185]: @Arkq Alright, can you file a followup issue to enable on Darwin too and I will take a look after this lands?

Comment (Arkq) [reply .github/workflows/tests.yaml:185]: OK, I've submitted a new issue for that: https://github.com/project-chip/connectedhomeip/issues/15905

**Pull request merged**